gateway-service/src/main/java/org/zowe/apiml/gateway/ribbon/RequestContextUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.ribbon;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.zuul.context.RequestContext;
+
+import java.util.Optional;
+
+/**
+ * Class for storing and retrieving instance information from RequestContext
+ */
+public class RequestContextUtils {
```

Comment:
Have you considered implementing as a Bean instead of as a collection of static methods?

Reply:
It's an utility class which by definitions contain static methods. What would be the advantage of creating instead a bean? Just asking, I don't know, we can definitely do it if it's worthy! :)

Reply:
I have considered it but decided to go with static class instead. Given the RequestContext is also static and threadlocal based and ease of testing it, i chose this simple way.

Reply:
Beans that we are using and most of the beans in general are Singletons and as such they are more or less equal to the Class with static methods. 

The difference is mainly in the usage, where the static classes are a bit simpler to use but more difficult to work with in tests. 

And in this case I am ok with both choices, I am happy with the answer from David.